internal/commands/kubernetes/create.go

```diff
@@ -0,0 +1,249 @@
+		SSHKeys:     sshKeys,
+		Storage:     p.storage,
+		KubeletArgs: kubeletArgs,
+		Taints:      taints,
+	}
+
+	return ng, nil
+}
+
+func processKubeletArg(in string) (upcloud.KubernetesKubeletArg, error) {
+	split := strings.SplitN(in, "=", 2)
+	if len(split) < 2 {
+		return upcloud.KubernetesKubeletArg{}, fmt.Errorf("invalid kubelet-arg: %s", in)
+	}
+
+	return upcloud.KubernetesKubeletArg{
+		Key:   split[0],
+		Value: split[1],
+	}, nil
+}
+
+func processLabel(in string) (upcloud.Label, error) {
+	split := strings.SplitN(in, "=", 2)
+	if len(split) < 2 {
+		return upcloud.Label{}, fmt.Errorf("invalid label: %s", in)
+	}
+
+	return upcloud.Label{
+		Key:   split[0],
+		Value: split[1],
+	}, nil
+}
+
+func processTaint(in string) (upcloud.KubernetesTaint, error) {
+	r := regexp.MustCompile(`^(.+)=(.+):(.+)`)
+	s := r.FindStringSubmatch(in)
+	if len(s) < 4 {
+		return upcloud.KubernetesTaint{}, fmt.Errorf("invalid taint: %s", in)
+	}
+
+	return upcloud.KubernetesTaint{
+		Effect: upcloud.KubernetesClusterTaintEffect(s[3]),
+		Key:    s[1],
+		Value:  s[2],
+	}, nil
+}
+
+type createCommand struct {
+	*commands.BaseCommand
+	params createParams
+	completion.Kubernetes
+}
+
+// InitCommand implements Command.InitCommand
+func (c *createCommand) InitCommand() {
+	fs := &pflag.FlagSet{}
+	c.params = createParams{CreateKubernetesClusterRequest: request.CreateKubernetesClusterRequest{}}
+
+	fs.StringVar(&c.params.Name, "name", "", "Kubernetes cluster name.")
+	fs.StringVar(&c.params.Network, "network", "", "Network to use. The value should be UUID of a private network.")
+	fs.StringVar(&c.params.NetworkCIDR, "network-cidr", "", "CIDR of the network being used.")
+	fs.StringArrayVar(
+		&c.params.nodeGroups,
+		"node-group",
+		[]string{},
+		"Node group(s) for running workloads, multiple can be declared.\n"+
+			"Usage: --node-group "+
+			"count=8,"+
+			"kubelet-arg=\"log-flush-frequency=5s\","+
+			"label=\"owner=devteam\","+
+			"label=\"env=dev\","+
+			"name=my-node-group,"+
+			"plan=K8S-2xCPU-4GB,"+
+			"ssh-key=\"ssh-ed25519 AAAAo admin@user.com\","+
+			"ssh-key=\"/path/to/your/public/ssh/key.pub\","+
+			"storage=01000000-0000-4000-8000-000160010100,"+
+			"taint=\"env=dev:NoSchedule\","+
+			"taint=\"env=dev2:NoSchedule\"",
+	)
+	fs.StringVar(&c.params.Zone, "zone", "", "Zone where to create the server.")
+	c.AddFlags(fs)
+
+	_ = c.Cobra().MarkFlagRequired("name")
+	_ = c.Cobra().MarkFlagRequired("network")
+	_ = c.Cobra().MarkFlagRequired("network-cidr")
+	_ = c.Cobra().MarkFlagRequired("node-group")
+	_ = c.Cobra().MarkFlagRequired("zone")
+}
+
+// ExecuteWithoutArguments implements commands.NoArgumentCommand
+func (c *createCommand) ExecuteWithoutArguments(exec commands.Executor) (output.Output, error) {
+	svc := exec.All()
+	msg := fmt.Sprintf("Creating cluster %s", c.params.Name)
+	exec.PushProgressStarted(msg)
+	if err := c.params.processParams(); err != nil {
+		return nil, err
+	}
+
+	exec.Debug("sending r", "params", c.params)
```

Comment:
Could make this more descriptive (e.g. `Creating kubernetes cluster, params:`) or remove

Reply:
Removed